wgengine/filter/filter.go

```diff
@@ -129,6 +131,79 @@ func maybeHexdump(flag RunFlags, b []byte) string {
 	return packet.Hexdump(b) + "\n"
 }
 
+// MatchesFromFilterRules parses a backward-compatible FilterRule used by control's
```

Comment:
// MatchesFromFilterRules parse a number of wire-format FilterRule values into
// the Matches format.
// If an error is returned, the Matches result is still valid, containing the rules that
// were successfully converted.